final/docker-compose.yml

```diff
@@ -0,0 +1,67 @@
+version: "3.8"
+services:
+  bitcoin-core:
+    build:
+      dockerfile: ./bitcoind/Dockerfile
+    image: bitcoind
+    container_name: bitcoind
+    command:
+      - --testnet
+      - -conf=/home/bitcoin/.bitcoin/bitcoin.conf
+      - -pid=/home/bitcoin/.bitcoin/testnet3/bitcoind-testnet.pid
+    volumes:
+      - data:/home/bitcoin/.bitcoin
+    expose:
+      - 8332
+      - 8333
+      - 18332
+      - 18333
+      - 18444
+    ports:
+      - 18332:18332
```

Comment:
I don't think we need to connect these ports to the host? also for the CLN section below

Reply:
so we don't need the `host` port mapping just `expose` them on network

Reply:
I don't understand your reply

Reply:
so, the `expose` field will just expose them on network made by compose for other services on network to access. the container to host port mapping is what you wanted me to get rid off. But won't I be needing to expose these to host as well so that other nodes can find me.

Reply:
you only need to connect the bitcoin p2p port to the host, not the RPC port, etc.

Reply:
so 8333, 18333 and 38333